internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -0,0 +1,117 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task-process-file-updates
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Update files in a Git repository
+  params:
+      - name: fileUpdates
+        type: string
+        description: fileUpdates json string
+  results:
+      - name: fileUpdatesInfo
+        description: fileUpdates detailed information
+      - name: fileUpdatesState
+        description: fileUpdates state
+  workspaces:
+      - name: temp
+        description: Temporary working directory
+  steps:
+      - name: perform-updates
+        image: quay.io/hacbs-release/release-utils@sha256:b850b1a4e93d16050eefdb967b7875c17d7af4e86ecb0b268674dc6c3e8d90b1
+        env:
+          - name: GITLAB_HOST
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_host
+          - name: ACCESS_TOKEN
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_access_token
+          - name: GIT_AUTHOR_NAME
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_name
+          - name: GIT_AUTHOR_EMAIL
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_email
+        script: |
+            #!/usr/bin/env sh
+            set -eo pipefail
+
+            # loading git and gitlab functions
+            . /home/utils/gitlab-functions
+            . /home/utils/git-functions
+
+            gitlab_init
+
+            # saves the fileUpdates raw json to a file
+            TEMP=$(mktemp).json
+            cat > $TEMP <<JSON
+            $(params.fileUpdates)
+            JSON
+
+            REPOSITORY=$(jq -cr '[.repo, .ref] | join(" ")' $TEMP)
```

Comment:
Wait, so the `fileUpdates` is not the whole `fileUpdates` section from the CR, it's just a subset for one repo? Then the description should make it clear.

Reply:
the input of this will indeed be different after our discussion yesterday in the Stand Up. David made the adjusts required in the `internal-request` script and I will implement this as it should. Working on it atm.

Reply:
ok, cool. Can you maybe provide more details on the expected format of `paths` in the parameter description? It could even include a sample snippet.

Reply:
sure. I will do.

Reply:
done.

Reply:
thanks